agent/taskresource/grpcclient/credentialsfetcherclient_test.go

```diff
@@ -0,0 +1,148 @@
+}
+
+func (*mockCredentialsFetcherServer) DeleteKerberosLease(ctx context.Context, req *pb.DeleteKerberosLeaseRequest) (*pb.DeleteKerberosLeaseResponse, error) {
+	if len(req.GetLeaseId()) == 0 {
+		return &pb.DeleteKerberosLeaseResponse{}, status.Errorf(codes.InvalidArgument, "credentialspecs request should not be empty")
+	}
+
+	return &pb.DeleteKerberosLeaseResponse{LeaseId: leaseid, DeletedKerberosFilePaths: []string{"/var/credentials-fetcher/krbdir/123456/webapp01", "/var/credentials-fetcher/krbdir/123456/webapp02"}}, nil
+}
+
+func dialer() func(context.Context, string) (net.Conn, error) {
+	listener := bufconn.Listen(1024 * 1024)
+
+	server := grpc.NewServer()
+
+	pb.RegisterCredentialsFetcherServiceServer(server, &mockCredentialsFetcherServer{})
+
+	go func() {
+		if err := server.Serve(listener); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	return func(context.Context, string) (net.Conn, error) {
+		return listener.Dial()
+	}
+}
+
+func TestCredentialsFetcherClient_AddKerberosLease(t *testing.T) {
+	tests := []struct {
+		name             string
+		credspecContents []string
+		response         CredentialsFetcherResponse
+	}{
+		{
+			"invalid request empty credspec contents",
+			[]string{},
+			CredentialsFetcherResponse{},
+		},
+		{
+			"valid request credspecs associated to gMSA account",
+			[]string{credspec_webapp01},
+			CredentialsFetcherResponse{LeaseId: leaseid, KerberosTicketPaths: []string{"/var/credentials-fetcher/krbdir/123456/webapp01", "/var/credentials-fetcher/krbdir/123456/webapp02"}},
+		},
+	}
+
+	ctx := context.Background()
+
+	conn, err := grpc.DialContext(ctx, "", grpc.WithInsecure(), grpc.WithContextDialer(dialer()))
+
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer conn.Close()
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			response, _ := NewCredentialsFetcherClient(conn, time.Minute).AddKerberosLease(context.Background(), tt.credspecContents)
+
+			if response.LeaseId != tt.response.LeaseId {
+				t.Error("error: expected", tt.response, "received", response)
+			}
+		})
+	}
+}
+
+func TestCredentialsFetcherClient_DeleteKerberosLease(t *testing.T) {
+	tests := []struct {
+		name     string
+		leaseid  string
+		response CredentialsFetcherResponse
+	}{
+		{
+			"invalid request empty leaseid input",
+			"",
+			CredentialsFetcherResponse{},
+		},
+		{
+			"valid request credspecs associated to gMSA account",
+			leaseid,
+			CredentialsFetcherResponse{LeaseId: leaseid, KerberosTicketPaths: []string{"/var/credentials-fetcher/krbdir/123456/webapp01", "/var/credentials-fetcher/krbdir/123456/webapp02"}},
+		},
+	}
+
+	ctx := context.Background()
+
+	conn, err := grpc.DialContext(ctx, "", grpc.WithInsecure(), grpc.WithContextDialer(dialer()))
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer conn.Close()
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			response, _ := NewCredentialsFetcherClient(conn, time.Minute).DeleteKerberosLease(context.Background(), tt.leaseid)
```

Comment:
same comment as for add-kerberos test above

Reply:
fixed comment